app/services/invoices/create_service.rb

```diff
@@ -12,6 +12,9 @@ def initialize(subscriptions:, timestamp:)
     end
 
     def create
+      invoice = nil
+      result = nil
```

Comment:
Actually, self.result is already defined so, I suggest to only do:

```ruby
  ActiveRecord::Base.transaction do
        result.invoice = Invoice.create!(
   # ...
```
and add a `delegate :invoice, to: :result` in the private section of the class

Reply:
that works for me!